tests/e2e/config_secret.go

```diff
@@ -206,11 +212,355 @@ var _ = ginkgo.Describe("Config-Secret", func() {
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Try to create a PVC and verify it gets bound successfully")
+		pvclaim2, err := createPVC(client, namespace, nil, "", storageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		var pvclaims2 []*v1.PersistentVolumeClaim
+		pvclaims2 = append(pvclaims2, pvclaim2)
+		ginkgo.By("Waiting for all claims to be in bound state")
+		_, err = fpv.WaitForPVClaimBoundPhase(client, pvclaims2, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			ginkgo.By("Deleting the PVC")
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim2.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Restart CSI controller pod")
+		err = updateDeploymentReplicawithWait(client, 0, vSphereCSIControllerPodNamePrefix, csiNamespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = updateDeploymentReplicawithWait(client, csiReplicas, vSphereCSIControllerPodNamePrefix, csiNamespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Try to create a PVC verify that it is stuck in pending state")
+		pvclaim3, err := createPVC(client, namespace, nil, "", storageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By("Expect claim status to be in Pending state")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimPending, client,
+			pvclaim3.Namespace, pvclaim3.Name, framework.Poll, time.Minute)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+			fmt.Sprintf("Failed to find the volume in pending state with err: %v", err))
+		defer func() {
+			ginkgo.By("Deleting the PVC")
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim3.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Update vsphere-config-secret file with testuser1 updated credentials")
+		createCsiVsphereSecret(client, ctx, configSecretUser1Alias, testUser1NewPassword, csiNamespace,
+			vCenterIP, vCenterPort, "")
+		defer func() {
+			ginkgo.By("Reverting back csi-vsphere.conf with its original vcenter user " +
+				"and its credentials")
+			createCsiVsphereSecret(client, ctx, vCenterUIUser, vCenterUIPassword, csiNamespace,
+				vCenterIP, vCenterPort, "")
+			revertOriginalvCenterUser = true
+
+			ginkgo.By("Restart CSI driver")
+			restartSuccess, err = restartCSIDriver(ctx, client, namespace, csiReplicas)
+			gomega.Expect(restartSuccess).To(gomega.BeTrue(), "csi driver restart not successful")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Restart CSI driver")
+		restartSuccess, err = restartCSIDriver(ctx, client, namespace, csiReplicas)
+		gomega.Expect(restartSuccess).To(gomega.BeTrue(), "csi driver restart not successful")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Wait for few mins for csi controller pods to be in running state")
+		time.Sleep(60 * time.Second)
```

Comment:
Instead of hard sleep  we can use this method 
err = fpod.WaitForPodsRunningReady(client, csiSystemNamespace, int32(num_csi_pods), 0,
		pollTimeout, ignoreLabels)
	gomega.Expect(err).NotTo(gomega.HaveOccurred())